strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/services/impl/ArtifactEntryServiceImpl.java

```diff
@@ -85,24 +106,28 @@ private boolean artifactEntryIsSavedForTheFirstTime(ArtifactEntry artifactEntry)
     {
         ArtifactCoordinates coordinates = entity.getArtifactCoordinates();
         Assert.notNull(coordinates, "coordinates should not be null");
-        
+
         ArtifactTag lastVersionTag = artifactTagService.findOneOrCreate(ArtifactTagEntry.LAST_VERSION);
-        Set<ArtifactTag> tagSet = new HashSet<>();
-        tagSet.add(lastVersionTag);
-        
-        Map<String, String> coordinatesMap = coordinates.dropVersion();
-        
-        List<ArtifactEntry> lastVersionArtifactList = findArtifactList(entity.getStorageId(),
-                                                                       entity.getRepositoryId(),
-                                                                       coordinatesMap, tagSet,
-                                                                       0,
-                                                                       -1,
-                                                                       "uuid", true).stream()
-                                                                                    .map(e -> (ArtifactEntry) getDelegate().detachAll(e,
-                                                                                                                                      true))
-                                                                                    .collect(Collectors.toList());
-        ArtifactEntry lastVersionEntry = lastVersionArtifactList.stream().findFirst().orElse(entity);
-        Optional<ArtifactCoordinates> lastVersionCoordinates = Optional.ofNullable(lastVersionEntry.getArtifactCoordinates());
+
+        entity.getArtifactGroups()
+              .stream()
+              .forEach(artifactGroup ->
+                       {
+                           ArtifactEntry lastVersionEntry = findLastVersionArtifactEntry(artifactGroup, lastVersionTag,
+                                                                                         entity);
+                           setLastVersionTag(lastVersionEntry, entity, lastVersionTag);
+                       }
+              );
+    }
+
+    private <S extends ArtifactEntry> void setLastVersionTag(ArtifactEntry lastVersionEntry,
+                                                             S entity,
+                                                             ArtifactTag lastVersionTag)
+    {
+        Optional<ArtifactCoordinates> lastVersionCoordinates = Optional.ofNullable(
```

Comment:
`artifactCoordinates` shouldn't be `null`

Reply:
it's from [master](https://github.com/strongbox/strongbox/blob/master/strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/services/impl/ArtifactEntryServiceImpl.java#L105)